Source/Diagnostics/ReducedDiags/LoadBalanceCosts.cpp

```diff
@@ -0,0 +1,194 @@
+
+    // keep track of the max number of boxes, this is needed later on to fill
+    // the jagged array (in case each step does not have the same number of boxes)
+    m_nBoxesMax = std::max(m_nBoxesMax, nBoxes);
+
+    // resize and clear data array
+    m_data.resize(m_nDataFields*nBoxes, 0.0);
+    m_data.assign(m_nDataFields*nBoxes, 0.0);
+
+    // read in WarpX costs_heuristic to local copy
+    amrex::Vector<std::unique_ptr<amrex::Vector<amrex::Real> > > costs_heuristic;
+    if (warpx.load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Heuristic)
+    {
+        costs_heuristic.resize(nLevels);
+        for (int lev = 0; lev < nLevels; ++lev)
+        {
+            costs_heuristic[lev].reset(new amrex::Vector<Real>);
+            const int nBoxesLev = warpx.getCostsHeuristic(lev)->size();
+            costs_heuristic[lev]->resize(nBoxesLev);
+        }
+        warpx.ComputeCostsHeuristic(costs_heuristic);
+    }
+
+    // keeps track of correct index in array over all boxes on all levels
+    int shift = 0;
+
+    // save data
+    for (int lev = 0; lev < nLevels; ++lev)
+    {
+        const amrex::DistributionMapping& dm = warpx.DistributionMap(lev);
+        const MultiFab & Ex = warpx.getEfield(lev,0);
+        for (MFIter mfi(Ex, false); mfi.isValid(); ++mfi)
+        {
+            const Box& tbx = mfi.tilebox();
+            m_data[shift + mfi.index()*m_nDataFields + 0] = (*costs_heuristic[lev])[mfi.index()];
+            m_data[shift + mfi.index()*m_nDataFields + 1] = dm[mfi.index()];
+            m_data[shift + mfi.index()*m_nDataFields + 2] = lev;
+            m_data[shift + mfi.index()*m_nDataFields + 3] = tbx.loVect()[0];
+            m_data[shift + mfi.index()*m_nDataFields + 4] = tbx.loVect()[1];
+            m_data[shift + mfi.index()*m_nDataFields + 5] = tbx.loVect()[2];
+        }
+
+        // we looped through all the boxes on level lev, update the shift index
+        shift += m_nDataFields*(costs_heuristic[lev]->size());
+    }
+
+    // parallel reduce to IO proc and get data over all procs
+    ParallelDescriptor::ReduceRealSum(m_data.data(), m_data.size(), ParallelDescriptor::IOProcessorNumber());
+
+    /* m_data now contains up-to-date values for:
+     *  [[cost, proc, lev, i_low, j_low, k_low] of box 0 at level 0,
+     *   [cost, proc, lev, i_low, j_low, k_low] of box 1 at level 0,
+     *   [cost, proc, lev, i_low, j_low, k_low] of box 2 at level 0,
+     *   ...
+     *   [cost, proc, lev, i_low, j_low, k_low] of box 0 at level 1,
+     *   [cost, proc, lev, i_low, j_low, k_low] of box 1 at level 1,
+     *   [cost, proc, lev, i_low, j_low, k_low] of box 2 at level 1,
+     *   ......] */
+
+}
+// end void LoadBalanceCosts::ComputeDiags
+
+// write to file function for cost
+void LoadBalanceCosts::WriteToFile (int step) const
+{
+    ReducedDiags::WriteToFile(step);
+
+    // get WarpX class object
+    auto& warpx = WarpX::GetInstance();
+
+    if (ParallelDescriptor::IOProcessor())
+    {
```

Comment:
```suggestion
    // get WarpX class object
    auto& warpx = WarpX::GetInstance();

    if (!ParallelDescriptor::IOProcessor()) return;
```
That will avoid 1 level of indentation in the rest of the function. Does that make sense?

Reply:
OK, indentation level is removed.